### PR TITLE
Switch from single topic for fixed delays to single topic for same intervals in Spring Kafka retry configuration.

### PR DESCRIPTION
**Pull request by Abhijeetmishr**

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ public RetryTopicConfiguration kafkaRetryTopicConfiguration(KafkaTemplate<?, ?>
 		KafkaProperties.Retry.Topic retryTopic = this.properties.getRetry().getTopic();
 		RetryTopicConfigurationBuilder builder = RetryTopicConfigurationBuilder.newInstance()
 			.maxAttempts(retryTopic.getAttempts())
-			.useSingleTopicForFixedDelays()
+			.useSingleTopicForSameIntervals()
```
Comment (garyrussell) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java:165]: Need to remove the `@SuppressWarnings` with this change.

**Review by garyrussell — CHANGES_REQUESTED** [review]

**Comment by scottfrederick** [comment]
This is blocked pending the `main` branch upgrading to Spring Kafka 3.0.4.

**Comment by Abhijeetmishr** [comment]
> This is blocked pending the `main` branch upgrading to Spring Kafka 3.0.4.

@scottfrederick blocked because of release can I make changes which is required to fix this issue ?

**Comment by Abhijeetmishr** [comment]
@garyrussell @scottfrederick PR is still needed or I should close it ?

**Comment by scottfrederick** [comment]
@Abhijeetmishr This is still a change we want to make, it's just not our highest priority. I'll merge this PR soon.

**Comment by Abhijeetmishr** [comment]
> @Abhijeetmishr This is still a change we want to make, it's just not our highest priority. I'll merge this PR soon.

I was confirming because I have to update this PR.
Thank you!

**Pull request closed without merging**